gapic/src/main/com/google/gapid/util/Logging.java

```diff
@@ -225,18 +224,28 @@ public void close() {
   /**
    * {@link Handler} that keeps a number of messages in memory for later retrieval.
    */
-  private static class MemoryHandler extends Handler {
-    private static final int MAX_SIZE = 10 * 1024 * 1024;
-
+  private static class Buffer extends Handler {
     public Runnable listener;
-    private final StringBuilder buffer = new StringBuilder();
+    private final LinkedList<Log.Message> buffer = new LinkedList<>();
```

Comment:
buffer is shadowing Logging.buffer. I suggest renaming Logging.buffer to Logging.messageBuffer